Add 'init' option to docker_container module to support docker's `--init` option

suzuki-shunsuke opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Add 'init' option to [docker_container module](http://docs.ansible.com/ansible/latest/docker_container_module.html) to support docker's `--init` option.
[This option requires Docker API 1.25+](https://docs.docker.com/engine/reference/commandline/run/#options).

Fixes #30761

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME
<!--- Name of the module, plugin, module or task -->
[docker_container module's](http://docs.ansible.com/ansible/latest/docker_container_module.html) `init` option

##### ANSIBLE VERSION

```
v2.4
```

suzuki-shunsuke committed:
Add support for `--init` option to `docker_container` module

Fixes #30761

Akasurde commented:
@suzuki-shunsuke Could you please rebase this branch ?

suzuki-shunsuke commented:
@Akasurde 
I have rebased this branch.

maxamillion commented on lib/ansible/modules/cloud/docker/docker_container.py:
```diff
@@ -131,6 +131,12 @@
+  init:
+    description:
+      - Run an init inside the container that forwards signals and reaps processes.
+        This option requires Docker API 1.25+.
```
Please provide input sanity checking against the docker API version so that if an user attempts to use `init` and doesn't have a new enough version of the API, they don't get a stack trace but instead a proper ansible `fail_json` message.

suzuki-shunsuke replied:
@maxamillion 
Thank you for your review.
I have added the validation of docker API's version.

felixfontein replied:
@maxamillion @suzuki-shunsuke The validation check wasn't enough, as `init` is still passed on, which leads to an error (see #42162).

suzuki-shunsuke replied:
@felixfontein 
Oh, I'm sorry.
I understand what is wrong by seeing https://github.com/ansible/ansible/issues/42162#issuecomment-401587809 .

suzuki-shunsuke committed:
Validate docker API's version when docker_container's init option is True

https://github.com/ansible/ansible/pull/34547#pullrequestreview-122355244
https://docs.docker.com/engine/reference/commandline/run/#options

init option requires docker API 1.25+ .

suzuki-shunsuke committed:
Fix failure of sanity test
